### PR TITLE
Get endpoints nc games t3.5

### PR DESCRIPTION
**Pull request by be-part**
Added tests and endpoint for /api

### CODE REVIEW

File: db/controllers/categories.controllers.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
-const { fetchCategories } = require("../models/categories.models")
+const { fetchCategories, fetchEndpoints } = require("../models/categories.models")
+
+exports.getEndpoints= (request, response) => {
+    fetchEndpoints().then((endpoints) => {
+        response.status(200).send({ endpoints: endpoints })
```
Comment (Jodiech) [inline db/controllers/categories.controllers.js]: Great use of destructuring with a custom key here in order to keep the code super clean! 🚀 There is no need to invoke the model for this, as we are not querying the database.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -10,6 +10,51 @@ afterAll(() => {
     connection.end()
 })
 
+describe.only('GET /api', () => {
+    test('get status 200 response', () => {
+        return request(app).get('/api').expect(200);
+    });
+    test('returns object containing all of the available endpoints of the api', () => {
+        return request(app).get('/api').expect(200).then((response)=> {
+            expect(response.body.endpoints).toEqual({
```
Comment (Jodiech) [inline __tests__/app.test.js]: Good test here 😄 Just a suggestion but this test can be condensed by requiring in the endpoints.json at the top of your tests and writing this instead:
 **expect(response.body.endpoints).toEqual(endpointsJson);** let me know if you need more info about this!

File: db/models/categories.models.js [hunk 3]
```diff
@@ -1,6 +1,16 @@
 const connection = require('../connection')
+const fs = require("fs/promises");
+
+exports.fetchEndpoints = () => {
+    return fs.readFile('endpoints.json', 'utf-8').then((result) => {
+        const jsObject = JSON.parse(result)
+        return jsObject
+    })
+}
```
Comment (Jodiech) [inline db/models/categories.models.js]: There should not be a model for this endpoint as you are not querying the database. You can add this in the controller to just read the file.

**Review by Jodiech — COMMENTED** [review]
Hey 👋 Nice work so far,  however, there seems to be some misunderstanding of the task overall and I'd like to go over this with you to double-check you're understanding.

File: db/controllers/endpoints.controller.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const endpoints = require('../../endpoints.json')
+
+exports.getEndpoints= (request, response) => {
+    console.log(endpoints)
```
Comment (leekli) [inline db/controllers/endpoints.controller.js:4]: Uh oh, a wild console log has appeared

**Review by leekli — APPROVED** [review]
Happy with the 3.5 changes Bronwyn, thank you.

be careful with this PR though as you appear to have ticket 4s code in here too

**Pull request merged**